### PR TITLE
Added attention and biased loss

### PR DESCRIPTION
**Pull request by YotYot**

### CODE REVIEW

File: toxic_fool/models/toxicity_clasifier_keras.py [hunk 1]
```diff
@@ -94,7 +182,7 @@ def _build_graph(self):
 
         model = keras.Model(inputs=self._input_layer, outputs=self._output_layer)
         adam_optimizer = keras.optimizers.Adam(lr=1e-3, decay=1e-6, clipvalue=5)
-        model.compile(loss='binary_crossentropy', optimizer=adam_optimizer, metrics=['accuracy', 'ce'])
+        model.compile(loss=CustomLoss.binary_crossentropy_with_bias, optimizer=adam_optimizer, metrics=['accuracy', 'ce', CalcAccuracy.precision, CalcAccuracy.recall, CalcAccuracy.f1])
```
Comment (orgoro) [inline toxic_fool/models/toxicity_clasifier_keras.py]: consider moving all metrics to a function  - 
`metrics = self.get_metrics()` for readability

File: toxic_fool/models/toxicity_clasifier_keras.py [hunk 2]
```diff
@@ -7,10 +7,97 @@
...
+    @staticmethod
+    def f1(y_true, y_pred):
+        precision = CalcAccuracy.precision(y_true, y_pred)
+        recall = CalcAccuracy.recall(y_true, y_pred)
+        return 2 * ((precision * recall) / (precision + recall + K.epsilon()))
+
+class CustomLoss(object):
+    @staticmethod
+    def binary_crossentropy_with_bias(y_true,y_pred):
+        return K.mean(K.binary_crossentropy(y_true, y_pred), axis=-1) + 0.01*K.sum(y_true)
```
Comment (orgoro) [inline toxic_fool/models/toxicity_clasifier_keras.py]: add argument for weight_recall=0.01
` 0.01*K.sum(y_true)` you meant -> ` 0.01*K.sum(y_pred)`
and I think
` 0.01*K.sum(y_pred* y.pred)` is better

Comment (orgoro) [reply toxic_fool/models/toxicity_clasifier_keras.py]:
```
def binary_crossentropy_with_bias(y_true, y_pred, weight_recall):
      return K.mean(K.binary_crossentropy(y_true, y_pred), axis=-1) + 
              weight_recall*K.sum(y_true*y_pred)

```

File: toxic_fool/models/toxicity_clasifier_keras.py [hunk 3]
```diff
@@ -7,10 +7,97 @@
...
+            mask = K.cast(mask, K.floatx())
+            ai = ai * mask
+        att_weights = ai / (K.sum(ai, axis=1, keepdims=True) + K.epsilon())
+        weighted_input = x * K.expand_dims(att_weights)
+        result = K.sum(weighted_input, axis=1)
+        if self.return_attention:
+            return [result, att_weights]
+        return result
+
+    def get_output_shape_for(self, input_shape):
```
Comment (orgoro) [inline toxic_fool/models/toxicity_clasifier_keras.py]: get_output_shape()

File: toxic_fool/models/toxicity_clasifier_keras.py [hunk 4]
```diff
@@ -7,10 +7,97 @@
...
+        ai = K.exp(logits - K.max(logits, axis=-1, keepdims=True))
+
+        # masked timesteps have zero weight
+        if mask is not None:
+            mask = K.cast(mask, K.floatx())
+            ai = ai * mask
+        att_weights = ai / (K.sum(ai, axis=1, keepdims=True) + K.epsilon())
+        weighted_input = x * K.expand_dims(att_weights)
+        result = K.sum(weighted_input, axis=1)
+        if self.return_attention:
```
Comment (orgoro) [inline toxic_fool/models/toxicity_clasifier_keras.py]: bad practice return type changes
return it always if someone doesnt need it he wont use it

**Review by orgoro — CHANGES_REQUESTED** [review]

**Review by orgoro — APPROVED** [review]
